cwl/src/main/scala/cwl/CwlType.scala

```diff
@@ -31,6 +32,16 @@ object CwlType extends Enumeration {
   val String = Value("string")
   val File = Value("File")
   val Directory = Value("Directory")
+
+  // Callers expect the last component to match the basename if a basename is specified.
+  private [cwl] def tempFile(basename: Option[String], asDirectory: Boolean): files.File = {
```

Comment:
As a heads up this _might_ not be necessary anymore with https://github.com/broadinstitute/cromwell/pull/3697, I'll check after yours is merge though.